### DISCUSSION TITLE
Regular Dependency Update Cycle for Milvus

### DISCUSSION THREAD

**Discussion by nish112022** [milvus-io/milvus#38053] [Ideas & Feature requests]
**EOL**
End-of-life (EOL) refers to the point in time when an open-source software (OSS) project or a particular version of it will no longer be maintained or supported by the developers or the community. EOL software or outdated libraries no longer receive updates or patches, leaving them vulnerable to threats. Staying updated helps ensure security, compatibility, and performance improvements.

Now, milvus uses a lot of dependencies which we can see under go.mod ,go.sum and a lot of them are quite old [more than 6 months] and can be brought to the later versions.

But, this will keep on happening.So,I believe we should have some monthly/biweekly upgrade going on so that we try to be on the latest versions almost  all the time.

**One example of this is:**
We currently use github.com/casbin/casbin/v2 at version v2.44.2. The latest in this series is [v2.101.0](https://pkg.go.dev/github.com/casbin/casbin/v2@v2.101.0), released on Nov 16, 2024, which includes numerous improvements and bug fixes.

**Comment by xiaofan-luan** [comment]
@nish112022 
Agreed on that.
We don't have a regular update check but feel free to upgrade manually by submitting a PR.

@LoveEachDay maybe it's reasonable to add a check

**Comment by nish112022** [comment]
Alright, I will do this.Time to time
